concent_api/common/decorators.py

```diff
@@ -85,3 +89,32 @@ def decorated(*args: Any, **kwargs: Any) -> None:
                 return func(*args, **kwargs)
 
         return decorated
+
+
+class retry_on_database_error(object):
```

Comment:
I think this is an overkill. It's too generic. We don't need any generic code for retrying on database errors because there is only one, verify specific edge case where we might want to do it - a race condition between two processes trying to create an object. And even then, the retry is only a workaround for the fact that Django can't do this properly at the database level with `INSERT ... ON CONFLICT ...` yet. Basically, if we could use the full power of SQL, this case would not need to result in an exception.

Actually, I've just checked and it seems that we might be able to use `INSERT ... ON CONFLICT ...` now:
- [django-postgres-extra > Using the manager > Conflict handling](https://django-postgres-extra.readthedocs.io/manager/) - this library was created in 2017 and seems to support `on_conflict`.
- [Add ON CONFLICT IGNORE support to QuerySet.bulk_create()](https://code.djangoproject.com/ticket/28668) - Django is adding `on_conflict` to `bulk_create()`. Not sure about ordinary `create()`. But we may have this feature in some form when we decide to update to a new release.

So I think something like this is not necessary and not even well adapted to our use case:
- Name does not say we're only retrying on uniqueness violation.
- We don't need complex retry conditions. We only ever want to retry once and we'll never use a timeout.
- This looks like a decorator but we barely ever want to put it on the whole function. It would be much more useful to be able to use it as a context manager to put it only around the operations we want to retry.

Reply:
ok.. decorator removed.. And thanks for explanation,  i'm interested in this subject so I will read this carefully for sure :)